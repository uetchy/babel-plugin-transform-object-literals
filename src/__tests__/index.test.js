import plugin from '..'
import pluginTester from 'babel-plugin-tester'
import path from 'path'

pluginTester({
  plugin,
  snapshot: true,
  tests: [
    `const json = {
      items: {
        name: "Sun",
        where: 'Solar sytem',
      },
    }`,
    `const json = {
      items: {
        1: 5,
        100: 500,
      },
    }`,
    ,
    `const json = {
      items: [1,2,3]
    }`,
    ,
    `const json = [1, '1', 'a']`,
  ],
})
