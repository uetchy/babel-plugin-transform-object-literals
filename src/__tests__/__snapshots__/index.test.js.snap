// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-transform-object-literals 1. babel-plugin-transform-object-literals: 1. babel-plugin-transform-object-literals 1`] = `
"
const json = {
      items: {
        name: 'Sun',
        where: 'Solar sytem',
      },
    }

      ↓ ↓ ↓ ↓ ↓ ↓

const json = JSON.parse(\\"{\\\\\\"items\\\\\\":{\\\\\\"name\\\\\\":\\\\\\"Sun\\\\\\",\\\\\\"where\\\\\\":\\\\\\"Solar sytem\\\\\\"}}\\");
"
`;

exports[`babel-plugin-transform-object-literals 2. babel-plugin-transform-object-literals: 2. babel-plugin-transform-object-literals 1`] = `
"
const json = {
      items: {
        1: 5,
        100: 500,
      },
    }

      ↓ ↓ ↓ ↓ ↓ ↓

const json = JSON.parse(\\"{\\\\\\"items\\\\\\":{\\\\\\"1\\\\\\":5,\\\\\\"100\\\\\\":500}}\\");
"
`;

exports[`babel-plugin-transform-object-literals 3. babel-plugin-transform-object-literals: 3. babel-plugin-transform-object-literals 1`] = `
"
const json = {
      items: [1,2,3]
    }

      ↓ ↓ ↓ ↓ ↓ ↓

const json = JSON.parse(\\"{\\\\\\"items\\\\\\":[1,2,3]}\\");
"
`;
